src/utils/tiles-filtering/filter-tile.ts

```diff
@@ -0,0 +1,192 @@
+import { Tile3D } from "@loaders.gl/tiles";
+import { FiltersByAttribute } from "../../types";
+import { AttributeStorageInfo, I3SAttributeLoader } from "@loaders.gl/i3s";
+import { load } from "@loaders.gl/core";
+import { TypedArray } from "@loaders.gl/schema";
+
+type I3STileAttributes = Record<string, string[] | TypedArray | null>;
+
+/**
+ * Filter tile indices by attribute value
+ * @param tile - tile to be filtered
+ * @param filtersByAttribute - custom filters patameters
+ * @returns {Promise<{isFiltered: boolean; id: string}>} Result of the tile filtering - isFiltered: true/false and tile id
+ */
+export const filterTile = async (
+  tile: Tile3D,
+  filtersByAttribute: FiltersByAttribute | null
+): Promise<{ isFiltered: boolean; id: string }> => {
+  const result = { isFiltered: false, id: tile.id };
+
+  if (tile.content.customfilters !== filtersByAttribute) {
```

Comment:
We use `userData` field in the tile to handle application data. See `src/utils/debug/texture-selector-utils.ts`. Could we use similar approach here (`tile.userData.customFilters` of `tile.content.userData.customFilters)?

Reply:
Done